### PR TITLE
IGNITE-19593 Optimized update assignments routine to start multiple tables at the same time.

### PR DESCRIPTION
**Pull request by ibessonov**
https://issues.apache.org/jira/browse/IGNITE-19593

### CODE REVIEW

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java [hunk 1]
```diff
@@ -794,6 +792,7 @@ private CompletableFuture<?> updateAssignmentInternal(ConfigurationNotificationE
                 startGroupFut
                         .thenComposeAsync(v -> inBusyLock(busyLock, () -> {
                             try {
+                                //TODO This procedure takes 10 seconds if there's no majority online.
```
[the remark this answers is not in the document]
Comment (ibessonov) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:795]: Thanks! I will create the issue tomorrow

**Comment by ibessonov** [comment]
It's been decided to have additional assignments VV instead of having two VVs with the same map. This way we'll hopefully make the code more clear

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java [hunk 2]
```diff
@@ -1112,11 +1124,7 @@ private CompletableFuture<?> createTableLocally(long causalityToken, String name
         // TODO: IGNITE-19082 Need another way to wait for indexes
         table.addIndexesToWait(collectTableIndexIds(tblId));
 
-        tablesByIdVv.update(causalityToken, (previous, e) -> inBusyLock(busyLock, () -> {
-            if (e != null) {
-                return failedFuture(e);
-            }
-
+        Function<Map<Integer, TableImpl>, CompletableFuture<Map<Integer, TableImpl>>> createTableClosure = previous -> {
```
[the remark this answers is not in the document]
Comment (ibessonov) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java]: I reverted this change

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java [hunk 3]
```diff
@@ -1231,11 +1252,7 @@ private void dropTableLocally(long causalityToken, String name, int tblId, List<
                 removeStorageFromGcFutures[p] = mvGc.removeStorage(replicationGroupId);
             }
 
-            tablesByIdVv.update(causalityToken, (previousVal, e) -> inBusyLock(busyLock, () -> {
-                if (e != null) {
-                    return failedFuture(e);
-                }
-
+            Function<Map<Integer, TableImpl>, CompletableFuture<Map<Integer, TableImpl>>> dropTableClosure = previousVal -> {
```
[the remark this answers is not in the document]
Comment (ibessonov) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java]: I reverted this change

**Review by sashapolo — APPROVED** [review]

File: modules/core/src/test/java/org/apache/ignite/internal/causality/CompletableVersionedValueTest.java [hunk 4]
```diff
@@ -267,6 +267,18 @@ public void testDefaultValue() throws Exception {
...
+    @Test
+    public void testLatest() {
+        CompletableVersionedValue<Integer> vv = new CompletableVersionedValue<>();
+
+        assertEquals(-1, vv.latestCausalityToken());
+
+        vv.complete(1, 10);
+
+        assertEquals(10, vv.latest());
+        assertEquals(1, vv.latestCausalityToken());
```
Comment (denis-chudov) [inline modules/core/src/test/java/org/apache/ignite/internal/causality/CompletableVersionedValueTest.java:279]:
```suggestion
        assertEquals(1, vv.latestCausalityToken());
                
        vv.complete(2);
        assertEquals(10, vv.latest());
        assertEquals(2, vv.latestCausalityToken());
```

Comment (ibessonov) [reply modules/core/src/test/java/org/apache/ignite/internal/causality/CompletableVersionedValueTest.java:279]: Thank you!

**Review by denis-chudov — APPROVED** [review]

**Pull request merged**